services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_core.py

```diff
@@ -1,135 +1,113 @@
-import datetime
+import asyncio
 import json
 import logging
 from typing import Any
 
 import arrow
 from fastapi import FastAPI
-from models_library.users import UserID
 from prometheus_api_client import PrometheusConnect
 from simcore_service_resource_usage_tracker.modules.prometheus import (
     get_prometheus_api_client,
 )
 
-# This script assumes everywhere that the minimum granularity of the data is 1 minute
-# Setting it smaller is unreasonable at least for prometheus, as the scraping interval is apprx. eq. to 1 h
-
+from .models.resource_tracker_container import ContainerResourceUsage
+from .modules.db.repositories.resource_tracker import ResourceTrackerRepository
 
 _logger = logging.getLogger(__name__)
 
 
-def _assure_dict_entry_exists(
-    metric_data, max_values_per_docker_id, image, userid
-) -> None:
-    for metric in metric_data:
-        current_id = metric["metric"]["id"]
-        if current_id not in max_values_per_docker_id.keys():
-            max_values_per_docker_id[current_id] = {
-                "container_uuid": metric["metric"]["container_label_uuid"],
-                "cpu_seconds": 0,
-                "uptime_minutes": 0,
-                "nano_cpu_limits": 0,
-                "egress_bytes": 0,
-                "image": image,
-                "user_id": userid,
-            }
-
-
-async def _evaluate_service_resource_usage(
+# def get_nested_value(data: dict, keys: list[str]) -> int | None:
```

Comment:
Avoid commented code https://github.com/ITISFoundation/osparc-simcore/blob/bf838aa1a559f060428536b17ad05e5876129982/docs/coding-conventions.md#cc2-no-commented-code

Reply:
thanks i forgot to delete it :+1: